### ISSUE TITLE
Publish user guide

### ISSUE THREAD

**Issue by cbeams** [bisq-network/bisq#169]
**Update Oct 11th**: This work is done, just waiting for #91 to actually complete so it can be merged into master.

---
## Background

The Asciidoc-based user guide is being developed under #91. This issue deals with publishing the HTML generated during the build process to the web where folks can get convenient access—and so that Bitsquare can refer directly to the docs as an in-app help system.
## Proposal

Let's publish to a `docs.bitsquare.io` subdomain as opposed to something like `bitsquare.io/docs`. This will allow for maximum independence and flexibility between the main website (#84) (which will be WordPress-based at this point), and the generated documentation, which is purely static HTML and requires nothing more than a basic webserver and ssh/sftp-based deployment.
## URL scheme requirements
- ideally be HTTPs-based, though not critical to start
- allow for internationalization (even though we'll only have English to start)
- be version-aware, so as to allow for different versions of Bitsquare to be in simultaneous use while still being able to reference accurate documentation for each.
### Different types of guides
- https://docs.bitsquare.io/en/0.1.0/userguide
- https://docs.bitsquare.io/en/0.1.0/devguide
- https://docs.bitsquare.io/en/0.1.0/arbguide (arbitrator guide; just a suggestion at this point)
### Different versions
- https://docs.bitsquare.io/en/0.1.0/userguide
- https://docs.bitsquare.io/en/0.2.0/userguide
- https://docs.bitsquare.io/en/1.0.0/userguide
### Snapshot versions

Documentation will be published automatically from the Travis CI build, meaning that we can also have docs specific to the latest snapshot:
- https://docs.bitsquare.io/en/0.1.0-SNAPSHOT/userguide
### "Current" version

A `current` alias always points to the latest released version:
- https://docs.bitsquare.io/en/current/userguide (points to, e.g., https://docs.bitsquare.io/en/0.1.0/userguide)

/cc @btcll

**Comment by btcll** [comment]
@cbeams Quick question - is it worth keeping the docs on the www. subdomain as this will save us from purchasing two SSL Certificates? I can work around the Wordpress URL stuff pretty easily so if you told me you wanted https://www.bitsquare.io/docs/* then that would be no problem. 

Can I also suggest, even though it will be basic HTML, we reference the help pages without a .html extension? I can add a .htaccess file to handle the re-writing. This gives us flexibility later if the docs end up being PHP or using some other server side language. 

As a side note - I'm very happy to provide the hosting for the docs. subdomain too. I have cPanel/PHP/MySQL hosting on a VPS available. Let me know if that's suitable and who needs FTP logins and I'll set them up.

**Comment by cbeams** [comment]
> @cbeams Quick question - is it worth keeping the docs on the www. subdomain as this will save us from purchasing two SSL Certificates? I can work around the Wordpress URL stuff pretty easily so if you told me you wanted https://www.bitsquare.io/docs/* then that would be no problem.

I haven't looked lately, but I imagine a non-ecommerce wildcard cert for covering all of bitsquare.io wouldn't be much, would it? Docs don't strictly have to be behind SSL at this early stage, but nice to have and sends the right message.

In any case, I'd like to keep systems separate and independent wherever possible (although integrated from a L&F and link navigation point of view). There are a number of reasons for this, but I'll skip them to be brief. Happy to go into detail and discuss further though, if you like.

> Can I also suggest, even though it will be basic HTML, we reference the help pages without a .html extension?

Sure, that's my plan as well. Simple to implement too, given that the docs in their current form are just single pages. The idea is to deploy them in the following form `.../userguide/index.html`, such that they can be linked to simply as http://docs.bitsquare.io/en/0.1.0/userguide.

Sound good?

> As a side note - I'm very happy to provide the hosting for the docs. subdomain too. I have cPanel/PHP/MySQL hosting on a VPS available. Let me know if that's suitable and who needs FTP logins and I'll set them up.

That's great! Assuming Manfred is OK with a Docker-based approach, all we really need is a fresh Linux box/VM, ideally with it's own IP address so as to simplify the DNS arrangement and allow for the Docker container serving the docs to be mapped directly to port 80. Would that that be possible with your setup?

**Comment by btcll** [comment]
I'll look into the wildcard cert, should work just fine. I agree that having everything behind SSL is for the best. Sends the right message, helps with Google rankings, etc etc

The directory system with index.html and just linking to the directory is even better than my suggestion! Make it happen :)

@ManfredKarrer could you please point the A Record for docs.bitsquare.io to 179.43.134.87 for me?

@cbeams I'll email you (and CC @ManfredKarrer ) FTP details. Let me know if there are any problems. If it's not suitable we can find something else, but I'm confident the VPS I have will be fine for what you've requested.

**Comment by cbeams** [comment]
I've removed the "Steps" section from the description, as we've taken a quicker path (some of which was discussed via email).

We now have, courtesy of @btcll, a CentOS VPS for docs running at 179.43.134.87. Chris and manfred have the credentials.
- [x] @ManfredKarrer to assign an A record pointing docs.bitsquare.io to 179.43.134.87.
- [x] Chris to configure Gradle and Travis CI to generate and upload docs.

**Comment by cbeams** [comment]
Manfred, I've just assigned this to you, as you're on point for the DNS step (though I'm working on my tasks in parallel as well)

**Comment by ManfredKarrer** [comment]
@btcll, @cbeams: I just added the DNS entry for http://docs.bitsquare.io.
Great to see all the progress! 
I will let all that part up to you, I don't have strong preferences in that area...

**Comment by cbeams** [comment]
Nice. Resolves for me. Thanks Manfred.

```
$ dig docs.bitsquare.io
[...]
;; QUESTION SECTION:
;docs.bitsquare.io.             IN      A

;; ANSWER SECTION:
docs.bitsquare.io.      3578    IN      A       179.43.134.87
```

**Comment by btcll** [comment]
@cbeams created ~/public_html/index.php with a PHP header redirect to the Userguide. Hope this helps, but feel free to delete/change if it doesn't.

**Comment by ManfredKarrer** [comment]
@btcll: Looks good already, the only tiny wish I have is to change the red color of the headlines to something less alerting (Bitsquare blue or the like)...

**Comment by btcll** [comment]
@ManfredKarrer  I'm not sure what the CSS options are, but if @cbeams can give me some guidance I am happy to do a Bitsquare CSS document up for the docs. It looks like it is all inline CSS at the moment.

**Comment by btcll** [comment]
Update on the SSL - it's only going to cover www.bitsquare.io and bitsquare.io. We could have added docs.bitsquare.io for $99/year -or- paid $200+ for a wildcare SSL Certificate. I decided not to (we can always do this later). This means we need to migrate from docs.bitsquare.io to https://www.bitsquare.io/docs/

Is that okay by you @cbeams ? Or do you have a better suggestion?

My intention is to do the migration following the HTTPS being setup. I'll need to email you fresh FTP/SSH info. I'm not sure we ever got to the bottom of the SSH issues for you either so let me know on IRC and we can troubleshoot that further.

**Comment by cbeams** [comment]
> Update on the SSL...

Hey Lloyd, I responded to this at https://github.com/bitsquare/bitsquare/issues/171#issuecomment-58631126. Thanks.

**Comment by cbeams** [comment]
Hey @btcll and @ManfredKarrer—regarding CSS, I've created #179, so we can track that work independently.

**Comment by cbeams** [comment]
I'm closing this, given that we worked out publication of docs a while ago. Getting to a final draft of the doc for 0.1 (#91) still remains to be done (see my comments there).
